micrometer/src/main/java/feign/micrometer/MeteredClient.java

```diff
@@ -45,38 +45,82 @@ public MeteredClient(Client client, MeterRegistry meterRegistry, MetricName metr
-          .recordCallable(() -> {
-            Response response = client.execute(request, options);
-            meterRegistry.counter(
-                metricName.name("http_response_code"),
-                metricName.tag(
-                    template.methodMetadata(),
-                    template.feignTarget(),
-                    Tag.of("http_status", String.valueOf(response.status())),
-                    Tag.of("status_group", response.status() / 100 + "xx")))
-                .increment();
-            return response;
-          });
+      final Response response = client.execute(request, options);
+      countSuccessResponseCode(template, options, response.status());
+      final Timer timer = createSuccessTimer(template, options);
+      sample.stop(timer);
+      return response;
     } catch (FeignException e) {
-      meterRegistry.counter(
-          metricName.name("http_response_code"),
-          metricName.tag(
-              template.methodMetadata(),
-              template.feignTarget(),
-              Tag.of("http_status", String.valueOf(e.status())),
-              Tag.of("status_group", e.status() / 100 + "xx")))
-          .increment();
+      countErrorResponseCode(template, options, e.status(), e);
       throw e;
     } catch (IOException | RuntimeException e) {
+      sample.stop(createExceptionTimer(template, options, e));
       throw e;
     } catch (Exception e) {
+      sample.stop(createExceptionTimer(template, options, e));
       throw new IOException(e);
     }
   }
 
+  private void countSuccessResponseCode(RequestTemplate template,
+                                        Options options,
+                                        int responseStatus) {
+    final List<Tag> tags = extraSuccessTags(template, options);
+    tags.add(Tag.of("http_status", String.valueOf(responseStatus)));
+    tags.add(Tag.of("status_group", responseStatus / 100 + "xx"));
+    final Tags allTags = metricName
+        .tag(template.methodMetadata(), template.feignTarget(), tags.toArray(new Tag[] {}));
+    meterRegistry.counter(
+        metricName.name("http_response_code"),
+        allTags)
+        .increment();
+  }
+
+  private void countErrorResponseCode(RequestTemplate template,
+                                      Options options,
+                                      int responseStatus,
+                                      Exception e) {
+    final List<Tag> tags = extraExceptionTags(template, options, e);
+    tags.add(Tag.of("http_status", String.valueOf(responseStatus)));
+    tags.add(Tag.of("status_group", responseStatus / 100 + "xx"));
+    final Tags allTags = metricName
+        .tag(template.methodMetadata(), template.feignTarget(), tags.toArray(new Tag[] {}));
+    meterRegistry.counter(
+        metricName.name("http_response_code"),
+        allTags)
+        .increment();
+  }
+
+  private Timer createSuccessTimer(RequestTemplate template, Options options) {
```

Comment:
I really like the idea of moving Timer creation (and all other metrics) creation to methods...

Could you please make them `protected`, if anyone has a need for a really complex customization, they would be able to do it here

Reply:
Done.